### ISSUE TITLE
New Python Handler: `pydantic` attribute `_hypothesis_plugin` not found

### ISSUE THREAD

**Issue by SupImDos** [mkdocstrings/mkdocstrings#380]
Thanks very much for your hard work on the new Python handler, I've been following this project's development and love your work.
I wasn't exactly sure whether to raise this issue here, or in [griffe](https://github.com/mkdocstrings/griffe), so sorry if I ended up picking the wrong one.

## Description
* The new Python handler (with `griffe`) for `mkdocstrings` crashes when `pydantic` is referred to in a docstring:
  ```python
  """Module level docstring"""
  
  import pydantic
  
  def fn(arg: pydantic.BaseModel) -> None:
      """Function docstring
  
      Args:
          arg (pydantic.BaseModel): This is going to break!
      """
      # This bit doesn't matter
      print(arg)
  ```
* It appears to be a `pydantic` specific issue, as the error refers to its `_hypothesis_plugin`:
  * https://github.com/samuelcolvin/pydantic/blob/master/setup.py#L139
  * https://github.com/samuelcolvin/pydantic/blob/master/pydantic/_hypothesis_plugin.py

## Example
* I've created a repo with my minimal reproducible example: https://github.com/SupImDos/mkdocstrings-pydantic-example

## Output
```shell
$ mkdocs serve
INFO     -  Building documentation...
INFO     -  Cleaning site directory
ERROR    -  Error building page 'index.md': module 'pydantic' has no attribute '_hypothesis_plugin'
Traceback (most recent call last):
  File "/home/fakeusername/Desktop/example/.venv/lib/python3.10/site-packages/mkdocs_autorefs/plugin.py", line 88, in get_item_url
    url = self._url_map[identifier]
KeyError: 'pydantic'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/fakeusername/Desktop/example/.venv/bin/mkdocs", line 8, in <module>
    sys.exit(cli())
  File "/home/fakeusername/Desktop/example/.venv/lib/python3.10/site-packages/click/core.py", line 1128, in __call__
    return self.main(*args, **kwargs)
  File "/home/fakeusername/Desktop/example/.venv/lib/python3.10/site-packages/click/core.py", line 1053, in main
    rv = self.invoke(ctx)
  File "/home/fakeusername/Desktop/example/.venv/lib/python3.10/site-packages/click/core.py", line 1659, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/fakeusername/Desktop/example/.venv/lib/python3.10/site-packages/click/core.py", line 1395, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/fakeusername/Desktop/example/.venv/lib/python3.10/site-packages/click/core.py", line 754, in invoke
    return __callback(*args, **kwargs)
  File "/home/fakeusername/Desktop/example/.venv/lib/python3.10/site-packages/mkdocs/__main__.py", line 177, in serve_command
    serve.serve(dev_addr=dev_addr, livereload=livereload, **kwargs)
  File "/home/fakeusername/Desktop/example/.venv/lib/python3.10/site-packages/mkdocs/commands/serve.py", line 54, in serve
    config = builder()
  File "/home/fakeusername/Desktop/example/.venv/lib/python3.10/site-packages/mkdocs/commands/serve.py", line 49, in builder
    build(config, live_server=live_server, dirty=dirty)
  File "/home/fakeusername/Desktop/example/.venv/lib/python3.10/site-packages/mkdocs/commands/build.py", line 314, in build
    _build_page(file.page, config, doc_files, nav, env, dirty)
  File "/home/fakeusername/Desktop/example/.venv/lib/python3.10/site-packages/mkdocs/commands/build.py", line 220, in _build_page
    output = config['plugins'].run_event(
  File "/home/fakeusername/Desktop/example/.venv/lib/python3.10/site-packages/mkdocs/plugins.py", line 102, in run_event
    result = method(item, **kwargs)
  File "/home/fakeusername/Desktop/example/.venv/lib/python3.10/site-packages/mkdocs_autorefs/plugin.py", line 197, in on_post_page
    fixed_output, unmapped = fix_refs(output, url_mapper)
  File "/home/fakeusername/Desktop/example/.venv/lib/python3.10/site-packages/mkdocs_autorefs/references.py", line 186, in fix_refs
    html = AUTO_REF_RE.sub(fix_ref(url_mapper, unmapped), html)
  File "/home/fakeusername/Desktop/example/.venv/lib/python3.10/site-packages/mkdocs_autorefs/references.py", line 156, in inner
    url = url_mapper(unescape(identifier))
  File "/home/fakeusername/Desktop/example/.venv/lib/python3.10/site-packages/mkdocs_autorefs/plugin.py", line 94, in get_item_url
    new_identifiers = fallback(identifier)
  File "/home/fakeusername/Desktop/example/.venv/lib/python3.10/site-packages/mkdocstrings/handlers/base.py", line 408, in get_anchors
    anchors = handler.renderer.get_anchors(handler.collector.collect(identifier, fallback_config))
  File "/home/fakeusername/Desktop/example/.venv/lib/python3.10/site-packages/mkdocstrings_handlers/python/collector.py", line 69, in collect
    loader.load_module(module_name)
  File "/home/fakeusername/Desktop/example/.venv/lib/python3.10/site-packages/griffe/loader.py", line 131, in load_module
    top_module = self._load_module_path(package.name, package.path, submodules=submodules)
  File "/home/fakeusername/Desktop/example/.venv/lib/python3.10/site-packages/griffe/loader.py", line 287, in _load_module_path
    self._load_submodules(module)
  File "/home/fakeusername/Desktop/example/.venv/lib/python3.10/site-packages/griffe/loader.py", line 292, in _load_submodules
    self._load_submodule(module, subparts, subpath)
  File "/home/fakeusername/Desktop/example/.venv/lib/python3.10/site-packages/griffe/loader.py", line 303, in _load_submodule
    member_parent[subparts[-1]] = self._load_module_path(
  File "/home/fakeusername/Desktop/example/.venv/lib/python3.10/site-packages/griffe/loader.py", line 285, in _load_module_path
    module = self._inspect_module(module_name, module_path, parent)
  File "/home/fakeusername/Desktop/example/.venv/lib/python3.10/site-packages/griffe/loader.py", line 347, in _inspect_module
    module = inspect(
  File "/home/fakeusername/Desktop/example/.venv/lib/python3.10/site-packages/griffe/agents/inspector.py", line 92, in inspect
    ).get_module(import_paths)
  File "/home/fakeusername/Desktop/example/.venv/lib/python3.10/site-packages/griffe/agents/inspector.py", line 206, in get_module
    value = dynamic_import(import_path, import_paths)
  File "/home/fakeusername/Desktop/example/.venv/lib/python3.10/site-packages/griffe/importer.py", line 67, in dynamic_import
    value = getattr(value, part)
AttributeError: module 'pydantic' has no attribute '_hypothesis_plugin'
```

## Versions
As per the `pyproject.toml` poetry dependencies:
* `python = "^3.10"`
* `mkdocs-material = "^8.1.11"`
* `mkdocstrings = {extras = ["python"], version = "^0.18.0"}`
* `pydantic = "^1.9.0"`

I'll keep having a look into the issue and see if I find anything else useful.
Thanks! 😃

**Comment by SupImDos** [comment]
## Update
Ah, I see now.
In summary, `pydantic` contains two *extra* modules for external plugins, that aren't used at usual runtime:

* [`_hypothesis_plugin.py`](https://github.com/samuelcolvin/pydantic/blob/master/pydantic/_hypothesis_plugin.py) for `hypothesis`
* [`mypy.py`](https://github.com/samuelcolvin/pydantic/blob/master/pydantic/_hypothesis_plugin.py) for `mypy`

But, `pydantic` does not include `hypothesis` or `mypy` as dependencies.

When `griffe` tries to [dynamically load these modules](https://github.com/mkdocstrings/griffe/blob/master/src/griffe/importer.py#L57), they raise a `ModuleNotFoundError`, as they themselves cannot import `hypothesis` / `mypy` (if you don't have them installed).

This is probably more of an issue with `pydantic`, so not sure what the best way forward is.

**Comment by pawamoy** [comment]
Fixed in Griffe 0.11.7! Thanks again!

**Comment by SupImDos** [comment]
Thanks so much!
